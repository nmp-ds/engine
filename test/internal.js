import { setup } from './_helpers.js'
import { expect, test } from 'vitest'

setup()

test('it generates css based on warp tokens', async ({ uno }) => {
  const classes = ['i-bg-$foo-bar', 'i-text-$biz-baz', 'i-border-$wombat-llama']
  const anticlasses = ['i-bg-foos-bars', 'i-text-bizs-bazs', 'i-border-wombats-llamas', 'i-magnuseses']
  const { css } = await uno.generate([...classes, ...anticlasses])
  expect(css).toMatchSnapshot()
})
