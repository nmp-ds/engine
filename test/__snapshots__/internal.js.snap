// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`it generates css based on warp tokens 1`] = `
"/* layer: default */
.i-bg-\\\\$foo-bar{background-color:var(--w-foo-bar);}
.i-text-\\\\$biz-baz{color:var(--w-biz-baz);}
.i-border-\\\\$wombat-llama{border-color:var(--w-wombat-llama);}
.i-shadow-\\\\$shadow-buttongroup{box-shadow:var(--w-shadow-buttongroup);}
.i-shadow-\\\\$shadow-card{box-shadow:var(--w-shadow-card);}
.i-shadow-\\\\$shadow-tooltip{box-shadow:var(--w-shadow-tooltip);}
.i-border-b-\\\\$color-alert-info-border{border-bottom-color:var(--w-color-alert-info-border);}
.i-border-l-\\\\$color-alert-info-border{border-left-color:var(--w-color-alert-info-border);}
.i-border-r-\\\\$color-alert-info-border{border-right-color:var(--w-color-alert-info-border);}
.i-border-t-\\\\$color-alert-info-border{border-top-color:var(--w-color-alert-info-border);}"
`;
